### PR TITLE
docs: add basic documents

### PR DESCRIPTION
**Pull request by Harish-osmosys**
Add basic document for this 
- Setup document: tell about the project setup steps 
- Learning guide: layout the things that need to be learned
- Development-flow : talks about major component and workflow
- Architecture - talks about the architecture of the project 

## Summary by CodeRabbit

- **Documentation**
  - Added `architecture.md` detailing the project's architecture and development workflows.
  - Added `development-flow.md` explaining key blockchain development concepts and workflows.
  - Added `learning-guide.md` as a comprehensive guide for Blockchain and DApps Development.
  - Added `setup.md` with instructions for setting up the project for development.
  - Updated `README.md` with links to the newly added documentation files.

### CODE REVIEW

File: docs/architecture.md [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+
+## Architecture Diagram
+
+![Architecture diagram](./images/blockchain.drawio.png)
+
+## Components
+
+### 1. Frontend (Angular)
+
+#### Overview
```
[the remark this answers is not in the document]
Comment (Harish-osmosys) [reply docs/architecture.md:42]: it already have a blank line

File: docs/setup.md [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+   - Import 2-3 account using the private key of that account
+   - Check this for reference : [Import Account](https://support.metamask.io/hc/en-us/articles/360015489331-How-to-import-an-account)
+   - Make sure you Import Account #0 as by default when you deploy any contract it is used
+
+5. Deploy the contract into localhost
+
+   ```sh
+   npx hardhat ignition deploy ./ignition/modules/Tracking.js --network localhost
+   ```
+
```
[the remark this answers is not in the document]
Comment (Harish-osmosys) [reply docs/setup.md:50]: it already have blank lines surrounding it

File: docs/setup.md [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+Before setting up for development, ensure you have the following prerequisites with the specified versions:
+
+- **NVM (Node Version Manager):** Use NVM to manage Node.js versions.
+- **Node.js** Node.js v20.x or higher. Can be installed via `nvm` using `nvm install 20` and used with `nvm use 20`.
+- **Git:** Git v2.x or higher.
+- **MetaMask:** Chrome Extension to interact with web3 application.
+
+These prerequisites are essential for deploying and running application in an environment.
+
```
[the remark this answers is not in the document]
Comment (Harish-osmosys) [reply docs/setup.md:15]: It have blank line surrounding it

File: docs/learning-guide.md [hunk 4]
```diff
@@ -0,0 +1,67 @@
+## Learning Guide for Blockchain and DApps Development
+
+### 1. Introduction to Blockchain
+- **What is Blockchain?**
```
Comment (Harish-osmosys) [inline docs/learning-guide.md:5]: Resolved in https://github.com/OsmosysSoftware/supply-chain-blockchain/pull/5/commits/5f92228e033fe62aa43fb88573ba685c31ef2078

File: docs/learning-guide.md [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+
+- **Truffle (Alternative to Hardhat)**
+  - Understand how to use Truffle, another popular development framework.
+  - Resources: [Truffle Suite Documentation](https://trufflesuite.com/docs/truffle/)
+
+- **Brownie (Python-based alternative)**
+  - Explore Brownie, a Python-based development and testing framework for smart contracts.
+  - Resources: [Brownie Documentation](https://eth-brownie.readthedocs.io/en/stable/)
+
+### 5. Building and Testing DApps
```
[the remark this answers is not in the document]
Comment (Harish-osmosys) [reply docs/learning-guide.md:40]: Resolved in https://github.com/OsmosysSoftware/supply-chain-blockchain/pull/5/commits/5f92228e033fe62aa43fb88573ba685c31ef2078

File: docs/learning-guide.md [hunk 6]
```diff
@@ -0,0 +1,67 @@
+## Learning Guide for Blockchain and DApps Development
+
+### 1. Introduction to Blockchain
```
[the remark this answers is not in the document]
Comment (Harish-osmosys) [reply docs/learning-guide.md:3]: Resolved in https://github.com/OsmosysSoftware/supply-chain-blockchain/pull/5/commits/5f92228e033fe62aa43fb88573ba685c31ef2078

File: docs/learning-guide.md [hunk 7]
```diff
@@ -0,0 +1,67 @@
...
+- **What is Blockchain?**
+  - Learn about the fundamental concepts of blockchain technology, including decentralization, consensus mechanisms, and the structure of a blockchain.
+  - Resources: [Blockchain Basics](https://www.ibm.com/topics/what-is-blockchain), [Mastering Bitcoin by Andreas Antonopoulos](https://github.com/bitcoinbook/bitcoinbook)
+
+### 2. Smart Contracts
```
[the remark this answers is not in the document]
Comment (Harish-osmosys) [reply docs/learning-guide.md:9]: Resolved in https://github.com/OsmosysSoftware/supply-chain-blockchain/pull/5/commits/5f92228e033fe62aa43fb88573ba685c31ef2078

File: docs/learning-guide.md [hunk 8]
```diff
@@ -0,0 +1,67 @@
...
+### 6. Deployment and Monitoring
+- **Deploying Smart Contracts**
+  - Learn how to deploy smart contracts to test networks and main networks.
+  - Resources: [Deploying Contracts with Hardhat](https://hardhat.org/tutorial/deploying-to-a-live-network.html), [Deploying with Truffle](https://trufflesuite.com/docs/truffle/getting-started/deploying/)
+
+- **Monitoring and Managing DApps**
+  - Understand how to monitor and manage deployed DApps using tools like Etherscan, The Graph, and Dune Analytics.
+  - Resources: [Etherscan](https://etherscan.io/), [The Graph Documentation](https://thegraph.com/docs/), [Dune Analytics](https://dune.com/)
+
+### 7. Advanced Topics and Ecosystem
```
[the remark this answers is not in the document]
Comment (Harish-osmosys) [reply docs/learning-guide.md:64]: Resolved in https://github.com/OsmosysSoftware/supply-chain-blockchain/pull/5/commits/5f92228e033fe62aa43fb88573ba685c31ef2078

File: docs/setup.md [hunk 9]
```diff
@@ -0,0 +1,93 @@
...
+   - For more details refer [this](https://github.com/OsmosysSoftware/supply-chain-management/blob/main/docs/contract-deployment.md)
+
+6. Start the application
+
+   ```sh
+   ng serve
+   ```
+
+   keep the hardhat node runnig in separate terminal
```
[the remark this answers is not in the document]
Comment (Harish-osmosys) [reply docs/setup.md]: Resolved in https://github.com/OsmosysSoftware/supply-chain-blockchain/pull/5/commits/5f92228e033fe62aa43fb88573ba685c31ef2078

File: docs/setup.md [hunk 10]
```diff
@@ -0,0 +1,93 @@
...
+## Prerequisites
+
+Before setting up for development, ensure you have the following prerequisites with the specified versions:
+
+- **NVM (Node Version Manager):** Use NVM to manage Node.js versions.
+- **Node.js** Node.js v20.x or higher. Can be installed via `nvm` using `nvm install 20` and used with `nvm use 20`.
+- **Git:** Git v2.x or higher.
+- **MetaMask:** Chrome Extension to interact with web3 application.
+
+These prerequisites are essential for deploying and running application in an environment.
```
[the remark this answers is not in the document]
Comment (Harish-osmosys) [reply docs/setup.md]: Resolved in https://github.com/OsmosysSoftware/supply-chain-blockchain/pull/5/commits/5f92228e033fe62aa43fb88573ba685c31ef2078

File: docs/setup.md [hunk 11]
```diff
@@ -0,0 +1,93 @@
...
+   ng serve
+   ```
+
+   keep the hardhat node runnig in separate terminal
+
+   Application will now be running locally at `http://localhost:4200/`.
+
+7. Add localhost network in metamask and switch to it
+
+8. Connect to Metamask wallet using your account
```
[the remark this answers is not in the document]
Comment (Harish-osmosys) [reply docs/setup.md:67]: Already have space in between

File: docs/setup.md [hunk 12]
```diff
@@ -0,0 +1,93 @@
...
+
+9.  Now you can interact with the application
+
+    ![Application](./images/app.png)
+
+## Docker Setup
+
+To run the application using a Docker container, follow these steps. Assuming the smart contract is already deployed on a public blockchain testnet (in this case, Sepolia testnet):
```
[the remark this answers is not in the document]
Comment (Harish-osmosys) [reply docs/setup.md:75]: not required in this case

File: docs/setup.md [hunk 13]
```diff
@@ -0,0 +1,93 @@
+# Development Setup
+
+This document outlines the steps required to set up this project for development. By following these steps, you'll be able to run your application locally with the necessary environment variables and database configuration.
+
+## Prerequisites
+
+Before setting up for development, ensure you have the following prerequisites with the specified versions:
+
+- **NVM (Node Version Manager):** Use NVM to manage Node.js versions.
+- **Node.js** Node.js v20.x or higher. Can be installed via `nvm` using `nvm install 20` and used with `nvm use 20`.
```
[the remark this answers is not in the document]
Comment (Harish-osmosys) [reply docs/setup.md:10]: not required

File: docs/setup.md [hunk 14]
```diff
@@ -0,0 +1,93 @@
...
+
+3. Start Hardhat node
+
+   ```sh
+   npx hardhat node
+   ```
+
+4. Import Account into metamask
+
+   - When you run the command mentioned in step 3 you will get list of 20 account with private key
```
[the remark this answers is not in the document]
Comment (Harish-osmosys) [reply docs/setup.md]: not required

File: docs/setup.md (hunk 12 above)
[the remark this answers is not in the document]
Comment (Harish-osmosys) [reply docs/setup.md:75]: not required

File: docs/setup.md [hunk 15]
```diff
@@ -0,0 +1,93 @@
...
+
+   - For more details refer [this](https://github.com/OsmosysSoftware/supply-chain-management/blob/main/docs/contract-deployment.md)
+
+6. Start the application
+
+   ```sh
+   ng serve
+   ```
+
+   keep the hardhat node running in a separate terminal
```
[the remark this answers is not in the document]
Comment (Harish-osmosys) [reply docs/setup.md:61]: already updated

File: docs/learning-guide.md [hunk 16]
```diff
@@ -0,0 +1,74 @@
...
+  - Learn about the fundamental concepts of blockchain technology, including decentralization, consensus mechanisms, and the structure of a blockchain.
+  - Resources: [Blockchain Basics](https://www.ibm.com/topics/what-is-blockchain), [Mastering Bitcoin by Andreas Antonopoulos](https://github.com/bitcoinbook/bitcoinbook)
+
+### 2. Smart Contracts
+
+- **Introduction to Smart Contracts**
+  - Understand what smart contracts are, how they work, and their significance in the blockchain ecosystem.
+  - Resources: [Ethereum Smart Contracts](https://ethereum.org/en/smart-contracts/), [Solidity Documentation](https://soliditylang.org/)
+
+### 3. Ethereum and Solidity
```
[the remark this answers is not in the document]
Comment (Harish-osmosys) [reply docs/learning-guide.md:15]: added

File: docs/learning-guide.md [hunk 17]
```diff
@@ -0,0 +1,74 @@
...
+
+- **Local Blockchain Development**
+  - Use tools like Ganache to create a personal blockchain for Ethereum development.
+  - Resources: [Ganache Documentation](https://www.trufflesuite.com/ganache)
+
+- **Testing Smart Contracts**
+  - Write tests for your smart contracts using tools like Mocha, Chai, and Waffle.
+  - Resources: [Testing with Hardhat](https://hardhat.org/guides/testing.html), [Testing with Truffle](https://trufflesuite.com/docs/truffle/testing/testing-your-contracts.html)
+
+### 6. Deployment and Monitoring
```
[the remark this answers is not in the document]
Comment (Harish-osmosys) [reply docs/learning-guide.md:54]: added

File: docs/learning-guide.md [hunk 18]
```diff
@@ -0,0 +1,74 @@
+## Learning Guide for Blockchain and DApps Development
+
+### 1. Introduction to Blockchain
+
+- **What is Blockchain?**
```
[the remark this answers is not in the document]
Comment (Harish-osmosys) [reply docs/learning-guide.md:5]: added

File: docs/setup.md (hunk 13 above)
[the remark this answers is not in the document]
Comment (Harish-osmosys) [reply docs/setup.md:10]: not required

File: docs/learning-guide.md [hunk 19]
```diff
@@ -0,0 +1,74 @@
...
+
+- **DeFi and NFTs**
+  - Learn about decentralized finance (DeFi) applications and non-fungible tokens (NFTs).
+  - Resources: [DeFi Pulse](https://defipulse.com/), [NFT School](https://nftschool.dev/)
+
+- **Security in Smart Contracts**
+  - Study best practices for securing smart contracts and common vulnerabilities.
+  - Resources: [Smart Contract Security by ConsenSys](https://consensys.net/diligence/blog/2020/09/contract-security-best-practices/), [OpenZeppelin Security](https://docs.openzeppelin.com/learn/developing-secure-contracts)
+
+This guide provides a structured approach to learning blockchain development, from understanding the basics to building and deploying DApps.
```
[the remark this answers is not in the document]
Comment (Harish-osmosys) [reply docs/learning-guide.md:74]: added

File: docs/architecture.md [hunk 20]
```diff
@@ -0,0 +1,93 @@
+# Project Architecture
```
Comment (xixas) [inline docs/architecture.md:1]: @Harish-osmosys  Update readme so users can navigate to these documents from ReadME. Users looking at readme won't know about these in current state

File: docs/architecture.md [hunk 21]
```diff
@@ -0,0 +1,93 @@
...
+
+## Overview
+
+This document provides an overview of the architecture of our blockchain project. The project consists of a frontend built with Angular, and smart contracts written in Solidity and managed with Hardhat.
```
Comment (xixas) [inline docs/architecture.md:5]: I would suggest adding a simple explaination about the current project here as example when explaining the architecture so users will have idea on what we are building and why we need to build it using blockchain and not just a normal DB.

**Review by xixas — CHANGES_REQUESTED** [review]

File: docs/setup.md (hunk 13 above)
[the remark this answers is not in the document]
Comment (Harish-osmosys) [reply docs/setup.md:10]: it fine

File: docs/setup.md [hunk 22]
```diff
@@ -0,0 +1,93 @@
...
+3. Start Hardhat node
+
+   ```sh
+   npx hardhat node
+   ```
+
+4. Import Account into metamask
+
+   - When you run the command mentioned in step 3 you will get a list of 20 accounts with private keys
+   - Import 2-3 account using the private key of that account
```
[the remark this answers is not in the document]
Comment (Harish-osmosys) [reply docs/setup.md:41]: it correct

File: docs/setup.md [hunk 23]
```diff
@@ -0,0 +1,93 @@
...
+
+   - For more details refer [this](https://github.com/OsmosysSoftware/supply-chain-management/blob/main/docs/contract-deployment.md)
+
+6. Start the application
+
+   ```sh
+   ng serve
+   ```
+
+   keep the hardhat node running in a separate terminal
```
[the remark this answers is not in the document]
Comment (Harish-osmosys) [reply docs/setup.md:61]: it fine

File: docs/setup.md [hunk 24]
```diff
@@ -0,0 +1,93 @@
...
+
+   make sure the hardhat node is running when you run this command
+
+   - For more details refer [this](https://github.com/OsmosysSoftware/supply-chain-management/blob/main/docs/contract-deployment.md)
+
+6. Start the application
+
+   ```sh
+   ng serve
+   ```
```
[the remark this answers is not in the document]
Comment (Harish-osmosys) [reply docs/setup.md:59]: it fine

File: docs/setup.md [hunk 25]
```diff
@@ -0,0 +1,93 @@
...
+   - When you run the command mentioned in step 3 you will get a list of 20 accounts with private keys
+   - Import 2-3 account using the private key of that account
+   - Check this for reference : [Import Account](https://support.metamask.io/hc/en-us/articles/360015489331-How-to-import-an-account)
+   - Make sure you Import Account #0 as by default when you deploy any contract it is used
+
+5. Deploy the contract into localhost
+
+   ```sh
+   npx hardhat ignition deploy ./ignition/modules/Tracking.js --network localhost
+   ```
```
[the remark this answers is not in the document]
Comment (Harish-osmosys) [reply docs/setup.md:49]: it fine

File: docs/setup.md [hunk 26]
```diff
@@ -0,0 +1,93 @@
...
+
+   ```sh
+   npm install
+   ```
+
+3. Start Hardhat node
+
+   ```sh
+   npx hardhat node
+   ```
```
[the remark this answers is not in the document]
Comment (Harish-osmosys) [reply docs/setup.md:36]: it fine

File: docs/setup.md [hunk 27]
```diff
@@ -0,0 +1,93 @@
...
+
+   ```sh
+   git clone https://github.com/OsmosysSoftware/supply-chain-blockchain
+   ```
+
+2. Install project dependencies:
+
+   ```sh
+   npm install
+   ```
```
[the remark this answers is not in the document]
Comment (Harish-osmosys) [reply docs/setup.md:30]: it fine

File: docs/setup.md [hunk 28]
```diff
@@ -0,0 +1,93 @@
...
+
+Please make sure to have these versions installed on your development server before proceeding with the setup.
+
+## Getting Started
+
+1. Clone the repository to your local machine:
+
+   ```sh
+   git clone https://github.com/OsmosysSoftware/supply-chain-blockchain
+   ```
```
[the remark this answers is not in the document]
Comment (Harish-osmosys) [reply docs/setup.md:24]: it fine

File: docs/architecture.md (hunk 21 above)
Comment (Harish-osmosys) [reply docs/architecture.md:5]: Updated in https://github.com/OsmosysSoftware/supply-chain-blockchain/pull/5/commits/f0a9eea9645c3517951ce350477fd46392f1bff9

File: docs/architecture.md (hunk 20 above)
Comment (Harish-osmosys) [reply docs/architecture.md:1]: updated in https://github.com/OsmosysSoftware/supply-chain-blockchain/pull/5/commits/f0a9eea9645c3517951ce350477fd46392f1bff9

**Review by xixas — APPROVED** [review]

**Pull request merged**